components/datepicker/DatePickerInput.js

```diff
@@ -0,0 +1,173 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Box from '../box';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import Icon from '../icon';
+import InputMetaPropTypes from '../input/InputMetaPropTypes';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames } from './utils';
+import { IconCalendarSmallOutline, IconWarningBadgedSmallFilled } from '@teamleader/ui-icons';
+import { TextSmall } from '../typography';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+import theme from './theme.css';
+
+class DatePickerInput extends PureComponent {
+  state = {
+    inputHasFocus: false,
+    selectedDate: null,
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    if (props.selectedDate !== undefined && props.selectedDate !== state.selectedDate) {
+      return {
+        selectedDate: props.selectedDate,
+      };
+    }
+
+    return null;
+  }
+
+  handleBlur = () => {
+    this.setState({ inputHasFocus: false });
+  };
+
+  handleFocus = () => {
+    this.setState({ inputHasFocus: true });
+  };
```

Comment:
You can just merge this in one handler again:

```js
handleFocus = () => {
  this.setState(({ inputHasFocus }) => { inputHasFocus: !inputHasFocus });
}
```

should be something like this. This handler can be a switch rather than two different handlers.

Reply:
You can say that for every switch then 😉